FreeRTOS_TCP_IP.c

```diff
@@ -1844,7 +1861,7 @@
 
                     if( pxSocket->u.xTCP.bits.bReuseSocket == pdFALSE_UNSIGNED )
                     {
-                        ( void ) FreeRTOS_closesocket( pxSocket );
+                        vSocketCloseNextTime( pxSocket );
```

Comment:
That would imply that this can indeed happen for orphaned (not yet accepted) sockets.
How about an assertion `configASSERT( xIsCallingFromIPTask() != pdFALSE );` here?

Reply:
Ah yes, we've just tested that either `bPassQueued` or `bPassAccept` is high, meaning that this is an orphaned socket.
Yes, then I like the configASSERT!

Reply:
Done.

Reply:
PS: I'm pretty sure we need to get rid of the calls into `vTCPStateChange()` from user task.
Especially the one originating from `prvTCPCreateStream()`
because it moves the socket from a connected to a disconnected state,
and thus calls the `xConnected->u.xTCP.pxHandleConnected( ( Socket_t ) xConnected, bAfter );`
which is will probably surprise the user callback by being called from a different task.